storage/proposal_signatures.go

```diff
@@ -0,0 +1,19 @@
+package storage
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/badger/transaction"
+)
+
+// ProposalSignatures represents persistent storage for proposer signatures on block proposals.
+type ProposalSignatures interface {
```

Comment:
As discussed above, I agree with Yurii that we should incorporate this into the `Headers` and `Blocks` storage interfaces rather than creating one just for signatures. 

I think we can also include some documentation there explaining why the signature storage is separate: 
> It is only transiently important conceptually (until the block obtains a QC), but our current business logic validates it even in cases where it is not strictly necessary. For simplicity, we require it be stored for all blocks, however it is stored separately to make it easier to remove in the future if/when we update the syncing and block ingestion logic.